src/main/java/com/internet/shop/security/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,23 @@
+package com.internet.shop.security;
+
+import com.internet.shop.exceptions.AuthenticationException;
+import com.internet.shop.lib.Inject;
+import com.internet.shop.lib.Service;
+import com.internet.shop.model.User;
+import com.internet.shop.service.UserService;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String login, String password) throws AuthenticationException {
+        User user = userService.findByLogin(login).orElseThrow(() ->
+                new AuthenticationException("Incorrect login!"));
+        if (user.getPassword().equals(password)) {
+            return user;
+        }
+        throw new AuthenticationException("Incorrect password");
+    }
```

Comment:
let's make both checks in one if and throw exception only once with message: incorrect login or password

Reply:
done